src/game/N3UIDebug.cpp

```diff
@@ -0,0 +1,209 @@
+#include "StdAfx.h"
+#include "N3UIDebug.h"
+
+#include "GameEng.h"
+#include "GameProcedure.h"
+#include "N3WorldManager.h"
+
+#include "N3Base/N3Camera.h"
+
+#include <imgui/imgui.h>
+#include <imgui/imgui_impl_dx9.h>
+#include <imgui/imgui_impl_win32.h>
+
+bool CN3UIDebug::s_bReleaseCalled = false;
+
+CN3UIDebug::CN3UIDebug() {}
+
+void CN3UIDebug::Release() {
+    s_bReleaseCalled = true;
+    ImGui_ImplDX9_Shutdown();
+    ImGui_ImplWin32_Shutdown();
+    ImGui::DestroyContext();
+}
+
+bool CN3UIDebug::Init() {
+    IMGUI_CHECKVERSION();
+    ImGui::CreateContext();
+    ImGuiIO & io = ImGui::GetIO();
+    io.ConfigFlags |= ImGuiConfigFlags_DockingEnable;
+    io.ConfigFlags |= ImGuiConfigFlags_ViewportsEnable;
+    io.ConfigFlags |= ImGuiConfigFlags_NoMouseCursorChange;
+    ImGui::StyleColorsDark();
+
+    ImGui_ImplWin32_Init(CN3Base::s_hWndBase);
+    ImGui_ImplDX9_Init(CN3Base::s_lpD3DDev);
+
+    ImGuiStyle & style = ImGui::GetStyle();
+    style.FrameRounding = 10;
+    style.Colors[ImGuiCol_Text] = ImVec4(0.90f, 0.98f, 0.81f, 1.00f);
+    style.Colors[ImGuiCol_WindowBg] = ImVec4(0.06f, 0.05f, 0.09f, 0.94f);
+    style.Colors[ImGuiCol_Border] = ImVec4(0.57f, 0.57f, 0.72f, 0.50f);
+    style.Colors[ImGuiCol_FrameBg] = ImVec4(0.22f, 0.32f, 0.69f, 0.54f);
+    style.Colors[ImGuiCol_TitleBg] = ImVec4(0.07f, 0.08f, 0.20f, 1.00f);
+    style.Colors[ImGuiCol_TitleBgActive] = ImVec4(0.17f, 0.19f, 0.40f, 1.00f);
+    style.Colors[ImGuiCol_Separator] = ImVec4(0.25f, 0.90f, 0.52f, 0.50f);
+    style.Colors[ImGuiCol_ResizeGrip] = ImVec4(0.16f, 0.91f, 0.27f, 0.20f);
+
+    return true;
+}
+
+void CN3UIDebug::BeginScene() {}
+
+void CN3UIDebug::EndScene() {
+    if (s_bReleaseCalled) {
+        return;
+    }
+
+    ImGui_ImplDX9_NewFrame();
+    ImGui_ImplWin32_NewFrame();
+    ImGui::NewFrame();
+
+    RenderDockingSpace();
+
+    static bool s_bDemoWindow = true;
+    ImGui::ShowDemoWindow(&s_bDemoWindow);
+
+    RenderGameMetrics();
+
+    ImGui::EndFrame();
+}
+
+void CN3UIDebug::Present() {
+    if (s_bReleaseCalled) {
+        return;
+    }
+
+    ImGui::Render();
+    ImGui_ImplDX9_RenderDrawData(ImGui::GetDrawData());
```

Comment:
Good point! Fixed here: https://github.com/ko4life-net/ko/pull/168/commits/008d11975754c02a2f7dc19ea55dbc30c567dd64

I think this might have fixed the issue I mentioned above, since I no longer experience it. Could you please check?

Reply:
I just pulled your changes and tried to reproduce. Seems to be that this did indeed fixed the issue. 🎉